lib/miq_expression/component/or.rb

```diff
@@ -0,0 +1,6 @@
+class MiqExpression::Component::Or < MiqExpression::Component::Composite
+  def to_arel(timezone)
+    first, *rest = sub_expressions
+    rest.inject(first.to_arel(timezone)) { |arel, sub_expression| arel.or(sub_expression.to_arel(timezone)) }
```

Comment:
@miq-bot yeah, I'm planning on doing something like this in a follow up to eliminate preprocessing. Right now we're going through the expression hash and pre-emptively pruning off anything that can't generate arel/sql before it even gets here.